[BUG]: TypeScript files do not find type declarations when `"moduleResolution": "node"` is used in tsconfig.json

### Description of the bug

As reported in [a previous issue](https://github.com/mheob/used-pm/issues/17), if the `moduleResolution` is `"node16"`, TypeScript ignores the `types` field.
However, if the `moduleResolution` is `"node"`, TypeScript will try to use only the `types` field instead of the `"types"` import condition.

It is necessary to include both the `"types"` import condition and the `types` field in the `package.json` file.

Sorry I didn't realize that at #17!

### Steps To Reproduce

0. Go into the new directory and create a `package.json` file with the following contents:

    ```json
    {}
    ```

2. Run `npm install used-pm@1.0.2 typescript`
3. Create a `tsconfig.json` file with the following contents:

    ```json
    {
      "compilerOptions": {
        "moduleResolution": "node",
        "esModuleInterop": true
      }
    }
    ```

4. Create a `foo.ts` file with the following contents:

    ```ts
    import usedPM from 'used-pm';
    console.log(usedPM());
    ```

5. Run `npx tsc --noEmit` -- This command should fail with the following error:

    ```console
    $ npx tsc --noEmit
    foo.ts:1:20 - error TS2307: Cannot find module 'used-pm' or its corresponding type declarations.
    
    1 import usedPM from 'used-pm';
                         ~~~~~~~~~
    
    
    Found 1 error in foo.ts:1
    ```

### Additional Information

+ You can also include the `main` field as a fallback for older Node.js.

    ```diff
     {
     	...
     	"exports": {
     		".": {
     			"import": {
     				"types": "./dist/index.d.mts",
     				"default": "./dist/index.mjs"
     			},
     			"require": {
     				"types": "./dist/index.d.cts",
     				"default": "./dist/index.cjs"
     			}
     		}
     	},
    +	"main": "./dist/index.cjs",
     	"types": "./dist/index.d.cts",
     	...
     }
    ```

    This is for Node.js 12.15 and earlier. Since [these versions are already EOL](https://github.com/nodejs/Release/tree/819f8073e42828b9a453b444d4a5a14da9d96844#end-of-life-releases) [^and-this-pkg-only-works-for-node14], there is no strong reason to support them.

    However, the `main` field is also necessary for some tools (e.g. old [Jest](https://jestjs.io/)) that look for modules without relying on Node.js.
    I think it would be better to include a `main` field.

    [^and-this-pkg-only-works-for-node14]: and the used-pm build results include the `??` operator, which is [only available in Node.js 14 or later](https://node.green/#ES2020-features--nullish-coalescing-operator-----)

+ For TypeScript 4.4 and below, the only type declaration file extension that can be specified in the `types` field is `.d.ts`. If you specify a file with a `.d.cts` extension in the `types` field, TypeScript 4.4 will throw an "TS2307: Cannot find module 'used-pm' or its corresponding type declarations." error.

    This means that *if you want to support TypeScript 4.4 or lower*, you need to change the extension of TypeScript files for CommonJS from `.cts` to `.ts`.

    However, TypeScript 4.4 will be EOL soon [^typescript-eol], so I don't think used-pm needs to support TypeScript 4.4 and below.
If you think you need it, change the file extension from `.cts` to `.ts`.

    [^typescript-eol]: Technically speaking, there is no EOL for TypeScript. We can still install TypeScript 1.0, and it is not [deprecated](https://docs.npmjs.com/deprecating-and-undeprecating-packages-or-package-versions/). However, one can speculate about the range of versions that should be supported.

        [DefinitelyTyped, the official repository for releasing `@types/*` packages](https://github.com/DefinitelyTyped/DefinitelyTyped), tests each package with every TypeScript released in the last two years.
        [TypeScript 4.4 will be two years old on August 26th of this year](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/acf0f103d8ebe423ae5794b3dd1e354d94dd5049/docs/support-window.json#L36), and will no longer be used for testing.

        I think this is like EOL.

+ I have tried [all the allowed values for `moduleResolution`](https://www.typescriptlang.org/tsconfig#moduleResolution): `"classic"`, `"node"`, `"node10"`, `"node16"`, `"nodenext"`, and `"bundler"`.
    When the `types` field is added, TypeScript will resolve type declarations correctly in all cases except `"classic"`!

    The [Classic module resolution](https://www.typescriptlang.org/docs/handbook/module-resolution.html#classic) algorithm is an old algorithm that does not use `package.json` at all, so I think used-pm does not need to support it.

--- a/package.json
+++ b/package.json
@@ -24,6 +24,7 @@
 			}
 		}
 	},
+	"types": "./dist/index.d.cts",
 	"files": [
 		"./dist/"
 	],

